src/cleanfigure.m

```diff
@@ -915,28 +1000,94 @@ function removeNaNs(meta, handle)
   end
 end
 % ==========================================================================
-function [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim)
-    % Determine the corners of the axes as defined by xLim and yLim
-    bottomLeft  = [xLim(1); yLim(1)];
-    topLeft     = [xLim(1); yLim(2)];
-    bottomRight = [xLim(2); yLim(1)];
-    topRight    = [xLim(2); yLim(2)];
+function [bottomLeft, topLeft, bottomRight, topRight] = corners2D(xLim, yLim)
+  % Determine the corners of the axes as defined by xLim and yLim
+  bottomLeft  = [xLim(1), yLim(1)];
+  topLeft     = [xLim(1), yLim(2)];
+  bottomRight = [xLim(2), yLim(1)];
+  topRight    = [xLim(2), yLim(2)];
+end
+% ==========================================================================
+function corners = corners3D(xLim, yLim, zLim)
+    % Determine the corners of the 3D axes as defined by xLim, yLim, and
+    % zLim
+
+    % Lower square of the cube
+    lowerBottomLeft  = [xLim(1), yLim(1), zLim(1)];
+    lowerTopLeft     = [xLim(1), yLim(2), zLim(1)];
+    lowerBottomRight = [xLim(2), yLim(1), zLim(1)];
+    lowerTopRight    = [xLim(2), yLim(2), zLim(1)];
+
+    % Upper square of the cube
+    upperBottomLeft  = [xLim(1), yLim(1), zLim(2)];
+    upperTopLeft     = [xLim(1), yLim(2), zLim(2)];
+    upperBottomRight = [xLim(2), yLim(1), zLim(2)];
+    upperTopRight    = [xLim(2), yLim(2), zLim(2)];
+
+    % Put the into one matrix
+    corners          = [lowerBottomLeft;
+                        lowerTopLeft;
+                        lowerBottomRight;
+                        lowerTopRight;
+                        upperBottomLeft;
+                        upperTopLeft;
+                        upperBottomRight;
+                        upperTopRight];
+end
+% ==========================================================================
+function P = getProjectionMatrix(meta)
+    % Calculate the projection matrix from a 3D plot into the image plane
+
+	% Get the projection angle
+    [az, el] = view(meta.gca);
+
+    % Convert from degrees to radians.
+    az = az*pi/180;
+    el = el*pi/180;
+
+    % The transformation into the image plane is done in a two-step process.
+    % First rotate around the z-axis by -az (in radians)
+    Rot_z = [ cos(-az)  -sin(-az)   0   0
+              sin(-az)   cos(-az)   0   0
+              0          0          1   0
+              0          0          0   1];
+
+    % Second rotate around the x-axis by (el - pi/2) radians.
+    % NOTE: There are some trigonometric simplifications, as we use
+    % (el-pi/2)
+    % cos(x - pi/2) =  sin(x)
+    % sin(x - pi/2) = -cos(x)
+    Rot_x = [ 1         0         0         0
+              0         sin(el)   cos(el)   0
+              0        -cos(el)   sin(el)   0
+              0         0        0          1];
+
+    % Get the data aspect ration
+    AspectRatio = get(meta.gca, 'DataAspectRatio');
+    scale       = 1./AspectRatio;
+    scaleMatrix = [ scale(1) 0        0        0
```

Comment:
You can write this more succinctly:
```matlab
scaleMatrix = diag([1./aspectRatio 1]);
```
Also, it is common to begin variable names (with more than 1 letter) with a lowercase letter, i.e. `rot_x`, `rot_z` and `aspectRatio`.